Userland/Applications/DisplaySettings/MonitorSettingsWidget.cpp

```diff
@@ -21,31 +21,46 @@ namespace DisplaySettings {
 
 MonitorSettingsWidget::MonitorSettingsWidget()
 {
+    m_screen_layout = GUI::WindowServerConnection::the().get_screen_layout();
     create_resolution_list();
     create_frame();
     load_current_settings();
 }
 
 void MonitorSettingsWidget::create_resolution_list()
 {
-    // TODO: Find a better way to get the default resolution
-    m_resolutions.append({ 640, 480 });
-    m_resolutions.append({ 800, 600 });
-    m_resolutions.append({ 1024, 768 });
-    m_resolutions.append({ 1280, 720 });
-    m_resolutions.append({ 1280, 768 });
-    m_resolutions.append({ 1280, 960 });
-    m_resolutions.append({ 1280, 1024 });
-    m_resolutions.append({ 1360, 768 });
-    m_resolutions.append({ 1368, 768 });
-    m_resolutions.append({ 1440, 900 });
-    m_resolutions.append({ 1600, 900 });
-    m_resolutions.append({ 1600, 1200 });
-    m_resolutions.append({ 1920, 1080 });
-    m_resolutions.append({ 2048, 1152 });
-    m_resolutions.append({ 2560, 1080 });
-    m_resolutions.append({ 2560, 1440 });
-    m_resolutions.append({ 3440, 1440 });
+    if (auto edid = EDID::Parser::from_framebuffer_device(m_screen_layout.screens[0].device, 0); !edid.is_error()) { // TODO: multihead
```

Comment:
What does this `TODO` mean? Could certainly be more descriptive.

Reply:
I'm not really sure, I used the same method to get the EDID as further down in the source file, so I felt there was value in maintaining the same comment so when that part is addressed both items can be found easily enough.

Reply:
Yeah, maintain the comment for the time being, someone might know what it means and be able to shed some light on it

Reply:
Multihead refers to multiple monitors. You use `screens[0]` here, so you are only retrieving resolutions for the first monitor.

Reply:
@GMTA is there a reason we do this? I envisioned a separate FramebufferDevice for each monitor that comes and goes as it is connected/disconnected, kind of like VirtIO.

Reply:
@tomuta no idea, just explaining the comment!